luigi/scheduler.py

```diff
@@ -655,9 +672,15 @@ def add_task(self, task_id=None, status=PENDING, runnable=True,
         if deps is not None:
             task.deps = set(deps)
 
+        if deps_configs is not None:
+            task.deps_configs = deps_configs
+
         if new_deps is not None:
             task.deps.update(new_deps)
 
+        if new_deps_configs is not None:
+            task.deps_configs.update(new_deps_configs)
```

Comment:
This looks plain wrong. Actually, I think this feature could be implemented totally agnostic of dynamic dependencies. Have you considered completely removing `new_deps_configs` and not thread it around? If you think it's necessary, can you point out a test case that would othewise fail without it?